apis/tempo/v1alpha1/tempostack_types.go

```diff
@@ -122,6 +122,10 @@ type TempoStackSpec struct {
 	// +kubebuilder:validation:Optional
 	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Observability"
 	Observability ObservabilitySpec `json:"observability,omitempty"`
+
+	// +required
+	// +kubebuilder:pruning:PreserveUnknownFields
+	ExtraConfig ConfigLayers `json:"extraConfig,omitempty"`
```

Comment:
Could this be `map[string]apiextensionsv1.JSON`?

https://pkg.go.dev/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1#JSON
https://github.com/prometheus-operator/prometheus-operator/blob/a8eb064ec76327afca43638085c82a20f6850689/pkg/apis/monitoring/v1beta1/alertmanager_config_types.go#L122

Reply:
I think we can use that structure